documentation/CanopyMatchMaker.md

```diff
@@ -1,3 +1,55 @@
-##UNTRUSTED COMPONENT##
+# Canopy Match Maker
+
+The canopy matchmaker uses three different metrics to decide the best configuration of the device
+* Explicit priorities from NP set, and implicit priorities from application specific settings
+* Canopy matching based on how closely the users preferences matches the capabilities of a solution.
+* Solution types
+
+Broadly, the match making algorithm used by the canopy match maker is as follows:
+1. Augmentation of the payload:
+    * Add inferred common terms to the preferences set
+    * Add information about solution types/categories to each solution
+    * Add an ontologized ISO24751 version of the NP set
+    * In the ISO-24751 ontology, calculate and add leaves (EL-paths) of preferences and of the capabilities of solutions
+3. Decide solution disposition for each context by:
+    * Further autgment solution entries with priority information:
+        * Extract explicit solution priorities from users NP set
+        * Add implicit solution priorities based on the presence of application specific settings in the user NP set
+    * Dispose solutions based on priority (see below for more details)
+    * For tied solutions, dispose based on canopy matching (see below for more details)
+
+More details are given below on solution types, as well as the two disposition methods used (priority and canopy).
+
+## Solution Types:
+To ensure that several conflicting solutions (e.g. two screenreaders) are not launched, "Types" are calculated for each solution, based on the capabilities they have (and in the future, on their explicitly stated types). The inference of solution type based on capabilities is based on the "apptology" tranformation, from the "flat" format into a solution type ontology. Generally, in the matchmaking process, each time a solution is 'accepted', any other conflicting solutions (i.e. of the same type) will be rejected.
+
+## Disposing from prioriy:
+Explicit priorities are declared in the NP sets metadata section, will always be 1024 or more. Implicit priorities are deduced from application specific settings. When a user has application specific settings for a solution, a priority of 512 is set for that solution.
```

Comment:
I dont think we ever actually made a decision on that... I've put floating point explicitly now, as that wont limit us wrt to the number of available priority values. Let me know if you think I should change it.